device/src/web.ts

```diff
@@ -110,6 +111,30 @@ export class DeviceWeb extends WebPlugin implements DevicePlugin {
       uaFields.operatingSystem = 'unknown';
     }
 
+    let match =
+      _ua.match(
+        /(opera|chrome|safari|firefox|msie|trident(?=\/))\/?\s*(\d+)/i,
+      ) || [];
+    let versions: RegExpMatchArray | RegExpExecArray | null;
+    if (/trident/i.test(match[1])) {
+      versions = /\brv[ :]+(\d+)/g.exec(_ua) || [];
+      uaFields.browserVersion = versions[1] || '';
+    } else if (match[1] === 'Chrome') {
+      versions = _ua.match(/\bOPR|Edge\/(\d+)/);
+      if (versions !== null) {
+        uaFields.browserVersion = versions[1] || '';
```

Comment:
should probably return `uaFields` here, otherwise the value gets overwritten on line 136
same for line 121, but I don't thing we support internet explorer, that's what trident stands for, right? that can probably be removed, thoughts @dwieeb?

should it try to get the full version instead of the major version as it does now? on iOS/Android it returns the full version

Reply:
Trident is IE yes, it might be some older versions of Edge too but I'd have to double check. If we don't support IE11 then we can remove that .

All of the major browsers are versioned using only the major version. Firefox, Chrome, and Edge don't use minor versions anymore when releasing new versions, it is just related to the build number. Really only the major version is somewhat useful. I am not sure about Safari, that might still have the minor version.

Reply:
It's not about usefulness but about consistency, since android and iOS return them.
If not useful, maybe android and iOS should return just the major?

Reply:
True, that makes sense. I will change web to include the minor version because it seems desktop and mobile Safari both still use the minor version pretty actively compared to other browsers